core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ast/planNodes/Sort.java

```diff
@@ -39,9 +40,25 @@ public Sort(PlanNode parent) {
 
 			Iterator<ValidationTuple> sortedTuplesIterator;
 
+			boolean closed = false;
+
 			@Override
 			public void close() throws SailException {
-				iterator.close();
+				if (closed) {
+					throw new IllegalStateException("Already closed");
+				}
+				closed = true;
+
+				if (sortedTuples != null && iterator.hasNext()) {
+					throw new AssertionError("All tuples from parent iterator where not retrieved when sorting!");
```

Comment:
This isn't right, we shouldn't be explicitly throwing AssertionErrors.

Reply:
My bad, I was doing this to make it easier to debug, will turn it back into an assert statement.

Reply:
Fixed it.